Agents/AQMeshInputAgent/AQMeshInputAgent/src/main/java/uk/ac/cam/cares/jps/agent/aqmesh/AQMeshInputAgent.java

```diff
@@ -394,6 +396,49 @@ private TimeSeries<OffsetDateTime> pruneTimeSeries(TimeSeries<OffsetDateTime> ti
         return new TimeSeries<>(newTimes, timeSeries.getDataIRIs(), newValues);
     }
 
+    public JSONArray calculateAndAddScaledMeasurementsForParticle(JSONArray readings) {
+        List<String> list = Arrays.asList("pm1", "pm2_5", "pm4", "pm10", "pm_tpc", "pm_total");
+        for (int i = 0; i < readings.length(); i++) {
+            JSONObject currentEntry = readings.getJSONObject(i);
+            for (int j = 0; j < list.size(); j++) {
+                String key = list.get(j);
+                Object prescaled = currentEntry.get(key + "_prescale");
+                Object slope = currentEntry.get(key + "_slope");
+                Object offset = currentEntry.get(key + "_offset");
+                if (prescaled == JSONObject.NULL || slope == JSONObject.NULL || offset == JSONObject.NULL) {
```

Comment:
You can check whether a JSON has a key using `has`, instead of checking for nulls

Reply:
The key will definitely be there but there is the possibility of the value being null (sensor spoils etc). If any of the value is null due to the above reasons, the agent will just allocate a null value to the scaled parameter instead.

Reply:
A cleaner way is to use `currentEntry.isNull(key + "_prescale")`, more of a personal preference